maltoolbox/language/languagegraph.py

```diff
@@ -116,9 +138,48 @@ def get_all_superassets(self) -> list[LanguageGraphAsset]:
             superassets.extend(current_asset.super_assets)
         return superassets
 
+
+    def get_associations(self) -> dict[str, LanguageGraphAssociation]:
+        """
+        Return a list of all of the associations that belong to this asset
+        directly or indirectly via inheritance.
+
+        Return:
+        A list of all of the associations that apply to this asset, either
+        directly or via inheritance.
+        """
+
+        associations = dict(self.associations)
```

Comment:
`self.associations` is a list. This will result in an error. For quicker lookup, wouldn't it make sense to have associations be a dict of some sort anyway?

Also, `get_associations` could be converted to an `@property` for a simpler interface to the language graph.

Reply:
The idea is to have the asset only track the associations that are directly specified on it and not the inherited ones as well. And this method should return all of the associations including the inherited ones.